server/src/test/java/org/opensearch/identity/AuthenticationTokenHandlerTests.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.identity;
+
+import org.hamcrest.MatcherAssert;
+import org.opensearch.authn.AuthenticationToken;
+import org.opensearch.authn.HttpHeaderToken;
+import org.opensearch.test.OpenSearchTestCase;
+
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+
+public class AuthenticationTokenHandlerTests extends OpenSearchTestCase {
+
+    public void testShouldExtractBasicAuthTokenSuccessfully() {
+
+        // The auth header that is part of the request
+        String authHeader = "Basic YWRtaW46YWRtaW4="; // admin:admin
+
+        AuthenticationToken authToken = new HttpHeaderToken(authHeader);
+
+        org.apache.shiro.authc.AuthenticationToken shiroAuthToken = AuthenticationTokenHandler.extractShiroAuthToken(authToken);
```

Comment:
I think this should be a non-shiro UsernamePasswordToken object that is returned

Reply:
I have a question about this. What do we want to do when we are using anything other than Basic Auth? I have set up some Bearer Auth but if we are going to basically only ever return UsernamePasswordToken we will need to account for equating the results between the different types of tokens--unless Shiro can automatically determine the UsernamePassword given other information?

Reply:
This is specific to these tests only afaik. Generally it should return Shiro AuthenticationToken